### PR TITLE
FIX : When we use TTC, we must update subprice on order and propal as well as in invoices

### PR DESCRIPTION
**Pull request by atm-gauthier**

### CODE REVIEW

File: htdocs/comm/propal/class/propal.class.php [hunk 1]
```diff
@@ -575,7 +578,7 @@ function updateline($rowid, $pu, $qty, $remise_percent, $txtva, $txlocaltax1=0.0
 			$this->line->localtax1_type		= $localtaxes_type[0];
 			$this->line->localtax2_type		= $localtaxes_type[2];
             $this->line->remise_percent		= $remise_percent;
-            $this->line->subprice			= $pu;
+            $this->line->subprice			= ($this->type==2?-abs($pu_ht):$pu_ht);
```
Comment (marcosgdf) [inline htdocs/comm/propal/class/propal.class.php]: Why adding this new condition?

**Comment by atm-gauthier** [comment]
That's wat we can find on updateline function in facture.class.php
I developped a module to allow add price in TTC and without what i added on this commit, update line in TTC doesn't work on propal and order.

**Comment by eldy** [comment]
The property $this->type==2 does not exists on proposal so shoul dnot be set so i don't understand the change.
You want to include a negative price on a proposal ?

**Comment by atm-gauthier** [comment]
Oh sorry, in fact i made a mistake, Test "$this->type==2" is just for invoice, actually i just need to save $pu_ht in $this->line->subprice.

**Comment by eldy** [comment]
Thanks. It's better. However, can you still describe how to reproduce bug (this is a sensitive part of code so i want to be sure for test before commiting it)

**Comment by rdoursenaud** [comment]
IMHO squashing the commits would also be nice to not clutter our history with your mistakes.

**Comment by eldy** [comment]
It seems this PR was set with "Discussion" or "PR to fix" tag, but no answer nor correction was provided since a long time. So request is automatically closed.
Please reopened and add a comment if you think this is an error.

**Pull request closed without merging**